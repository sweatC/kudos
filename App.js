import React from 'react';
import { AsyncStorage, StyleSheet, Text, View } from 'react-native';
import RootNavigator from './navigation/root-navigator';
import ListOfUsersPatches from './components/list-of-users-patches';


export default class App extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			firstName: '',
			lastName: '',
			email: '',
			password: ''
		}
		this.setSource = this.setSource.bind(this);
	}
  	render() {
		const data = [{key: 1, txt: "Tnak You"}, {key: 2, txt: "Thanks"}, {key:3, txt: "glhf next"}];
			const uData = [
				{ 
					key: 1, 
					img: { uri: 'https://pbs.twimg.com/profile_images/831993825635745796/HnVmB0-k.jpg' },
					fullName: "Rexar Agro", 
					count: 23	
				},
				{
					key: 2,
					img: { uri: 'https://pbs.twimg.com/profile_images/831993825635745796/HnVmB0-k.jpg' },
					fullName: "Valeera Wise",
					count: 74
				},
				{
					key: 3,
					img: { uri: 'https://pbs.twimg.com/profile_images/831993825635745796/HnVmB0-k.jpg' },
					fullName: "Guldan Windless",
					count: 11
				}
			]
    	return(
			<ListOfUsersPatches data={uData}/>
      		//<RootNavigator screenProps={{state: this.state, setUser: this.setSource}}/>
      	);
  	}
	setSource(user) {
		this.setState({user});
		AsyncStorage.setItem("user", JSON.stringify(user));
	}
}
