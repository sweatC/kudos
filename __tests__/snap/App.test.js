import React from 'react';
import App from '../../App';
import renderer from 'react-test-renderer';


it('App renders without crashing', () => {
  const rendered = renderer.create(<App />).toJSON();
  expect(rendered).toMatchSnapshot();
});
