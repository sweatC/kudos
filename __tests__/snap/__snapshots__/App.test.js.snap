// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
    ]
  }
>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
          "flexDirection": "column-reverse",
        },
        Object {
          "backgroundColor": "#000",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        collapsable={undefined}
        pointerEvents="auto"
        style={
          Object {
            "backgroundColor": "#E9E9EF",
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "shadowColor": "black",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0.2,
            "shadowRadius": 5,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "flex": 4,
                "resizeMode": "center",
              }
            }
          />
          <View
            style={
              Object {
                "flex": 2,
                "marginTop": 40,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "hsla(52, 75%, 6%, 0.91)",
                  "fontFamily": "Roboto",
                  "fontSize": 28,
                  "fontWeight": "400",
                  "textAlign": "center",
                }
              }
            >
              Kudos
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "hsla(52, 75%, 6%, 0.91)",
                  "fontFamily": "Roboto",
                  "fontSize": 14,
                  "fontWeight": "100",
                  "marginTop": 2,
                  "textAlign": "center",
                }
              }
            >
              Your small gestures grtitude
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 3,
                "marginTop": 15,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "hsla(52, 75%, 6%, 0.91)",
                  "borderRadius": 5,
                  "height": 45,
                  "width": 250,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                onPress={[Function]}
                style={
                  Object {
                    "color": "#fff",
                    "marginTop": 12,
                    "textAlign": "center",
                  }
                }
              >
                Sign in
              </Text>
            </View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              onPress={[Function]}
              style={
                Object {
                  "color": "hsla(52, 75%, 6%, 0.91)",
                  "fontFamily": "Roboto",
                  "fontSize": 10,
                  "marginTop": 7,
                  "textAlign": "center",
                }
              }
            >
              Don't have an account? Sign up
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      cardStyle={undefined}
      collapsable={undefined}
      getScreenDetails={[Function]}
      headerMode={undefined}
      index={0}
      layout={
        Object {
          "height": 0,
          "initHeight": 0,
          "initWidth": 0,
          "isMeasured": false,
          "width": 0,
        }
      }
      leftInterpolator={[Function]}
      mode="float"
      navigation={
        Object {
          "dispatch": [Function],
          "goBack": [Function],
          "navigate": [Function],
          "setParams": [Function],
          "state": Object {
            "index": 0,
            "routes": Array [
              Object {
                "key": "Init-id-1515836293937-0",
                "routeName": "Welcome",
              },
            ],
          },
        }
      }
      rightInterpolator={[Function]}
      router={
        Object {
          "getActionForPathAndParams": [Function],
          "getComponentForRouteName": [Function],
          "getComponentForState": [Function],
          "getPathAndParamsForState": [Function],
          "getScreenConfig": [Function],
          "getScreenOptions": [Function],
          "getStateForAction": [Function],
        }
      }
      titleInterpolator={[Function]}
      transitionConfig={undefined}
    >
      <View
        collapsable={undefined}
        onLayout={[Function]}
        style={
          Object {
            "backgroundColor": "#F7F7F7",
            "borderBottomColor": "rgba(0, 0, 0, .3)",
            "borderBottomWidth": 0.5,
            "height": 64,
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 20,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "flexDirection": "row",
                },
              ]
            }
          >
            <View
              collapsable={undefined}
              pointerEvents="box-none"
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "bottom": 0,
                  "justifyContent": "center",
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <Text
                accessibilityTraits="header"
                accessible={true}
                allowFontScaling={true}
                collapsable={undefined}
                ellipsizeMode="tail"
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "rgba(0, 0, 0, .9)",
                    "fontSize": 17,
                    "fontWeight": "600",
                    "marginHorizontal": 16,
                    "textAlign": "center",
                  }
                }
              >
                Welcome
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
